network-upgrader/src/main.rs

```diff
@@ -255,19 +469,40 @@ async fn upgrade_consensus_parameters(
 async fn transfer(transfer: &Transfer) -> anyhow::Result<()> {
     let provider = Provider::connect(transfer.url.as_str()).await?;
     let consensus_parameters = provider.consensus_parameters().clone();
-    let secret_key = handle_secret()?;
-    let wallet = WalletUnlocked::new_from_private_key(secret_key, Some(provider));
+    let signer = handle_secret(transfer.aws_kms_key_id.clone()).await?;
+    let wallet =
+        Wallet::from_address(signer.get_address()?.into(), Some(provider.clone()));
```

Comment:
While your PR looks logical to Rust SDK, it seems we don't need it=) I think we can create a `KMSWallet` that implements `ViewOnlyAccount` and `Account`(implementation looks simple) and use this wallet=)

<img width="888" alt="image" src="https://github.com/user-attachments/assets/33c7f7f4-98f0-4b4c-a696-6bb4f9ad4cbf">

<img width="609" alt="image" src="https://github.com/user-attachments/assets/c774025e-35f3-4f5b-93a6-099702e3cb2a">

Reply:
It will be a full duplicate of code from `Account` and they accepted my PR to be on nightly.
I will do the change to speed up the process but I think it's too much code duplication

Reply:
Done :)